### DISCUSSION TITLE
Changes to config.json, and additional files to remedy deletion of config.json

### DISCUSSION THREAD

**Discussion by RollinJay** [ikemen-engine/Ikemen-GO#1145] [Feature Request] [answered]
### Is your feature request related to a problem? Please describe.

To solve some issues, config.json is necessary to modify from the user side. Creators have no real way to have default settings in their game after building the engine.  

Another issue is that config.json has lines that can really break games if messed with by end users.

### Describe the solution you'd like

The addition of a default_config.json in the exterenal/folder, so that if config.json is deleted, the game default to the creators base settings. 

This default_config.json could also contain settings that the creators do not want modified (frame rate, window icon, etc) 

### Search terms

Config, settings, defaults

**Comment by Kokorz** [comment]
Bumping this as it's extremely important for fullgames, and i really hope it gets addressed before an eventual 1.0 engine release. 

The executable's inbuilt default_config.ini is nice... but only if you set your game up Ikemen's way, which is massively inconvenient for those who actually set up their games properly so that they can update the engine without struggle. And god forbid someone deletes their config file for one reason or another and now the engine cant find their screenpack, or if it does open (a.k.a. their files were set wrong for a fullgame), the engine resolution doesn't match the screenpack's resolution, the common states and animations and sprites and what have you don’t load resulting in crashes, and whatever else the engine subjects you to as config.ini handles a ton.

If not an external default_config.ini that gets used instead of the internal one when detected, maybe make it so options.lua, which already also recalls default settings that are defined in the file, is the file where defaults are pulled from? everyone can modify options.lua, even though the Lua files are very volatile right now due to the upcoming refactor, so i don't think it's a bad choice.

if that's not an acceptable route, maybe make it so you can define defaults for the screenpack in system.def, then the engine scan for a system.def if no config file is made, and if it finds one, it pulls from that file to try and initialize config defaults before finally loading the game. If this route is taken, it would also be a bonus if setting parameters were able to set as unconfigurable by dev choice, meaning it wont be added as an editable parameter in config.ini, unless, say, the debug folder is created, or another parameter is set in that system.def. 

having to compile the engine just to change defaults is quite annoying... i hope i'm not the only one who feels that way

> ### Is your feature request related to a problem? Please describe.
> 
> To solve some issues, config.json is necessary to modify from the user side. Creators have no real way to have default settings in their game after building the engine.  
> 
> Another issue is that config.json has lines that can really break games if messed with by end users.
> 
> ### Describe the solution you'd like
> 
> The addition of a default_config.json in the exterenal/folder, so that if config.json is deleted, the game default to the creators base settings. 
> 
> This default_config.json could also contain settings that the creators do not want modified (frame rate, window icon, etc) 
> 
> ### Search terms
> 
> Config, settings, defaults

**Comment by K4thos** [comment, marked as answer]
done in #2828
